### DISCUSSION TITLE
Support for nest

### DISCUSSION THREAD

**Discussion by olawalejuwonm** [typegoose/typegoose#933] [Ideas]
I was wondering if this library could be the default for using mongoose in nestjs (https://docs.nestjs.com/)

I don't have any idea for this implementation but I will love to make contributions

**Comment by hasezoey** [comment]
there once existed [`nestjs-typegoose`](https://www.npmjs.com/package/nestjs-typegoose), but it didnt get any updates for 4 years now, though there exist active forks of it like [`@m8a/nestjs-typegoose`](https://www.npmjs.com/package/@m8a/nestjs-typegoose), though they are not from the typegoose maintainers.

i personally have no intention of adding nestjs capability directly to `@typegoose/typegoose`.

**Comment by smolinari** [comment]
@olawalejuwonm - I'm the maintainer of [@m8a/nestjs-typegoose`](https://www.npmjs.com/package/@m8a/nestjs-typegoose). I'm also a moderator in the [Nest community](https://discord.gg/CNgAKEgD). It isn't up to Typegoose to make it the default for Nest, but rather for Nest. I actually made that suggestion some time ago, but it got shot down. I even made PR to get Typegoose's usage in as a recipe, but that also got shot down. 

So, you can still use Typegoose with Nest and I continuously recommend it, because it is a number of steps above Nest's Mongoose module in terms of TypeScript support. I at least got the name in a tag in the Database Forum in the Nestjs Discord Server. 

![image](https://github.com/typegoose/typegoose/assets/3314957/207327bd-9984-40ff-be5b-f006988d6083)

But, Typegoose won't become the default for Mongoose in Nest any time soon. 

Scott

**Reply by olawalejuwonm** [reply]
I'm creating a ticket for this on nest

**Reply by olawalejuwonm** [reply]
https://github.com/nestjs/nest/issues/13704 We can at least try to know why it wasn't accepted

**Reply by smolinari** [reply]
Yeah. Thanks for trying. But, I fear you won't get any decent reply. 

Scott
